libs/shared/ui-elements/src/lib/prio-button/prio-button.component.ts

```diff
@@ -9,10 +11,20 @@ import { Component, OnInit, Input, Output, EventEmitter, HostBinding } from '@an
 
 export class PrioButtonComponent implements OnInit {
 
-  inputFocus = false;
+  @Input() style = 'solid' ;  //solid || outline || flat
+  @Input() type = 'btn-primary' ;  //btn-primary || btn-secondary || btn-danger, ...etc
+  @Input() size = 'normal' ;   //normal || small
+  @Input() isBlock = false ;  // if true: button take dislay as block
+  @Input() isIcon = false ;  // if true: turns button view to icon (also if no text exists) and it takes class btn-icon
+  @Input() label = '' ;    //buttontext
+  // @Input() iconShape = '';
+  @Input() loadingState = ClrLoadingState.DEFAULT ;  // loading state
```

Comment:
We should rely on Typescript Types  `loadingState:ClrLoadingState` and shouldn't have a default value as well

Reply:
why should not get a default state although it has a clrLoadingstate.Default by default?
@Abdallah-khalil

Reply:
i meant that a feature in clarity we may not need so we don't have to provide a default value